Keen.NetStandard.Test/EventCacheTest.cs

```diff
@@ -0,0 +1,110 @@
+            await cache.Add(new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_AddNotEmpty_Success(IEventCache cache)
+        {
+            await cache.Clear();
+            Assert.Null(await cache.TryTake());
+            await cache.Add(new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            Assert.NotNull(await cache.TryTake());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_AddClearEmpty_Success(IEventCache cache)
+        {
+            await cache.Add( new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            await cache.Clear();
+            Assert.Null(await cache.TryTake());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_Iterate_Success(IEventCache cache)
+        {
+            await cache.Clear();
+            await cache.Add( new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            await cache.Add( new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
+            Assert.NotNull(await cache.TryTake());
+            Assert.NotNull(await cache.TryTake());
+            Assert.Null(await cache.TryTake());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void CachingPCL_SendEmptyEvents_Success(IEventCache cache)
+        {
+            var client = new KeenClient(SettingsEnv, cache);
+            Assert.DoesNotThrow(client.SendCachedEvents);
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void CachingPCL_ClearEvents_Success(IEventCache cache)
+        {
+            var client = new KeenClient(SettingsEnv, cache);
+            Assert.DoesNotThrow(() => client.EventCache.Clear());
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void CachingPCL_AddEvents_Success(IEventCache cache)
+        {
+            var client = new KeenClient(SettingsEnv, cache);
+
+            Assert.DoesNotThrow(() => client.AddEvent("CachedEventTest", new { Property = "Value" }));
+            Assert.DoesNotThrow(() => client.AddEvent("CachedEventTest", new { Property = "Value" }));
+            Assert.DoesNotThrow(() => client.AddEvent("CachedEventTest", new { Property = "Value" }));
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task CachingPCL_SendEventsParallel_Success(IEventCache cache)
+        {
+            await cache.Clear();
+            var client = new KeenClient(SettingsEnv, cache);
+            if (UseMocks)
+                client.Event = new EventMock(SettingsEnv,
+                                             addEvents: (e, p) => new List<CachedEvent>());
+
+            (from i in Enumerable.Range(1,100)
+             select new { Property = "Value" })
```

Comment:
I reformatted the document using visual studio mac. Hopefully it looks a little better now. Would be nice to have some sort of style guide. I'd default to what most people do in C#.